File: web_backend/web_backend/Models/WasteType.cs
using System.ComponentModel.DataAnnotations;
namespace web_backend.Models
{

    public class WasteType
    {
        [Key]
        public int WasteTypeId { get; set; }

        [Required]
        [MaxLength(50)]
        public string TypeName { get; set; }

        [MaxLength(255)]
        public string? Description { get; set; }

        public int BasePoint { get; set; } = 0;

        [MaxLength(20)]
        public string? ColorCode { get; set; }

        [MaxLength(80)]
        public string? IconName { get; set; }

        public bool IsActive { get; set; } = true;
    }
}

File: mobil/EcoLink/app/src/main/java/com/ecolink/app/network/models/WasteType.java
package com.ecolink.app.network.models;

public class WasteType {
    private int wasteTypeId;
    private String typeName;
    private String description;
    private int basePoint;
    private String colorCode;
    private String iconName;

    public int getWasteTypeId() { return wasteTypeId; }
    public String getTypeName() { return typeName; }
    public int getBasePoint() { return basePoint; }
    public String getColorCode() { return colorCode; }
}
